fe/fe-core/src/main/java/com/starrocks/authentication/CustomAuthenticationProvider.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.authentication;
+
+import com.starrocks.mysql.MysqlPassword;
+import com.starrocks.mysql.privilege.AuthPlugin;
+import com.starrocks.mysql.privilege.Password;
+import com.starrocks.sql.ast.UserIdentity;
+
+public abstract class CustomAuthenticationProvider implements AuthenticationProvider {
```

Comment:
Hi @lianneli , i understand what you want to do, but seems that you don't need to add an extra `CustomAuthenticationProvider` class, `AuthenticationProvider` is designed to achieve that.

Reply:
If you want to add an extra authentication method, you can refer to `KerberosAuthenticationProvider`. It uses a standalone jar file to enable kerberos authentication for Sr.

Reply:
Thanks for your advice! I delete `CustomAuthenticationProvider` and add the Custom implementation for check password methods in `Password` class. If it's ok?

Reply:
@lianneli how do you specify the server address where you auth with when calling `customAuthenticationProvider.authenticate(remoteUser, "", remotePassword, randomString, null);`?

Reply:
For username which created and saved in StarRocks, the server address and other params should be setted in user's jar, for which StarRocks do not need to consider.
﻿
As for other usernames, I find that you have designed `create security integration` SQL syntax which can transfer params to Custom Class. Users can set config `authentication_chain` properly and get those params by the code: 
```java
CustomSecurityIntegration securityIntegration = (CustomSecurityIntegration) authenticationInfo.extraInfo.get("SECURITY_INTEGRATION"); 

Map<String, String> userParams = securityIntegration.getPropertyMap();

```
﻿
by the way, I want to complete the implementation about `Drop security integration` and `Alter security integration` recently, to make the second scenario more convenient to use.

Reply:
@lianneli SecurityIntegration should be used with Config.authentication_chain, and acutally SecurityIntegration is an enterprise feature. For some reason, we left some of the code in community edition, and these remain code will probably be removed in the future, so you can't rely on SecurityIntegration to implement what you want to do. Considering this situation, i think the best choice is to refer to `KerberosAuthenticationProvider` and maintain your own patch for now. Sorry for your time and we still hope for your contribution in the future.

Reply:
Oh, btw, if you have any interest of our enterprise edition, i can ask our pre-sales engineer to contact you.

Reply:
@nshangyiming It's really regrettable that you donot provide security integration related SQL syntax! I can only maintain my patch for company's scenario. 
Custom authentication for users in SR is OK?

Reply:
Yes, if you don't rely on `SecurityIntegration`, you can still consider how to implement a custom authenticator, and that is absolutely welcome. BTW, i think trino already has a good design: https://trino.io/docs/current/develop/password-authenticator.html , you can refer to that as a start point and consider how to integrate it with SR. Thanks again.

Reply:
@nshangyiming Thanks for your advice. I have changed the implementation with independent conf file, and do not dependency security integration. Could you please review again?